### PR TITLE
Add integration test for /deletechat API in deletechatTestcase branch

### PR DESCRIPTION
**Pull request by santhiya-mj**
Implement integration testing for the /deletechat API endpoint to verify retrieval of all chats, updating changes to the deletechatTestcase branch

### CODE REVIEW

File: test/IntegrationTests/test_deletechat.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        logger.info(f"OUTPUT: Chat creation response status: {chat_response.status_code}")
+
+        self.assertEqual(chat_response.status_code, 200)
+        chat_data = chat_response.json()
+        self.valid_chat_id = chat_data["chat_id"]
+        logger.info(f"OUTPUT: Received valid chat_id: {self.valid_chat_id}")
+        logger.info("=== setUp completed successfully ===\n")
+
+    def test_valid_customer_guid_and_valid_chat_id(self):
+        """Test case 1: Valid customer_guid and valid chat_id"""
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_deletechat.py:50]: In this testcase, you should add chats, then use getchats to check whether they exist and then delete the chats and use getchats to verify whether they exist or not

File: test/IntegrationTests/test_deletechat.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 200)  # Expecting 200 based on API behavior
+        logger.info("=== Test Case 6 Completed ===\n")
+
+    def test_wrong_customer_guid_and_wrong_chat_id(self):
+        """Test case 7: Invalid customer_guid and chat_id"""
+        logger.info("=== Starting Test Case 7: Wrong customer_guid and wrong chat_id ===")
+
+        url = f"{self.BASE_URL}/chat"
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_deletechat.py:182]: Wrong api call?

File: test/IntegrationTests/test_deletechat.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("customer_guid is not valid", response.text.lower())
+
+        logger.info("=== Test Case 7 Completed ===\n")
+
+    def test_correct_customer_guid_wrong_chat_id(self):
+        """Test case 8: Valid customer_guid but invalid chat_id"""
+        logger.info("=== Starting Test Case 8: Correct customer_guid and wrong chat_id ===")
+
+        url = f"{self.BASE_URL}/chat"
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_deletechat.py:204]: wrong api call?

File: test/IntegrationTests/test_deletechat.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("chat_id is not valid", response.text.lower())
+        logger.info("=== Test Case 8 Completed ===\n")
+
+    def test_wrong_customer_guid_correct_chat_id(self):
+        """Test case 9: Valid customer_guid but invalid chat_id"""
+        logger.info("=== Starting Test Case 9: Correct customer_guid and wrong chat_id ===")
+
+        url = f"{self.BASE_URL}/chat"
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_deletechat.py:225]: wrong api call?

File: test/IntegrationTests/test_deletechat.py (hunk 1 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_deletechat.py:49]: Done, sir. I added chat messages, retrieved all chats using GetAllChats API, deleted all chats, and then verified that the deleted chats no longer exist using GetAllChats API.

File: test/IntegrationTests/test_deletechat.py (hunk 2 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_deletechat.py]: Test case updated with corrected API call method, sir

File: test/IntegrationTests/test_deletechat.py (hunk 3 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_deletechat.py]: Test case updated with corrected API call method, sir.

**Pull request closed without merging**